CleverRaven/Cataclysm-DDA
# mdeath

#### Summary
Features "monster death effects use spells"


#### Purpose of change
Make it so monster death effects are spells which are json, instead of hardcoded.

#### Describe the solution
1) store mdeath data into mtype. i made a new class for this that holds a fake_spell, message, and "corpse type", which has the "regular" death effects that should stay hardcoded: splatter, normal, and broken.
2) call the fake spell and messages properly in monster::die. 
3) create new spell effects that fill in the gaps so all monster death effects can be spells, except the few that are weird enough to delete (jabberwocky) or adjust (amigara)
4) change all monster json to have new json. this was accomplished with a combination of a new monster_death.json file with spells, and judicious application of find and replace with regular expressions (so i could remove the 59 occurences of `"death_function": [ "NORMAL" ]` as that is the default now)

#### Testing
1) use debug menu to kill a zombie dog to make sure the NORMAL death effect works properly.
2) spawn a `dog` to see if the thing spawns properly when it dies.
3) spawn a gasoline zombie and acidic ant to see the fire and acid fields spawn properly.
4) go to a triffid heart to see that the timed_event actually gets called.
5) i suppose you could do more, but that seemed like a good enough variety to prove to me the concept works.

#### Additional context
!WARNING! Any pull requests that add new monsters will cause merge conflicts. even if it doesn't show; it'll make the game not run due to the death function not being read into json properly.
